rust/Earthfile

```diff
@@ -56,16 +52,18 @@ CARGO:
     ARG --required args
     ARG output
     DO +SET_CACHE_MOUNTS_ENV
-    IF [ "$EARTHLY_KEEP_FINGERPRINTS" = "false" ]
-       DO +REMOVE_SOURCE_FINGERPRINTS
-    END
     RUN --mount=$EARTHLY_RUST_CARGO_HOME_CACHE --mount=$EARTHLY_RUST_TARGET_CACHE \
       set -e; \
+      echo "+CARGO: cargo $args"; \
       cargo $args; \
+      echo "+CARGO: sweeping target cache"; \
       cargo sweep -r -t $EARTHLY_SWEEP_DAYS; \
       cargo sweep -r -i; \
-      $EARTHLY_FUNCTIONS_HOME/copy-output.sh "$output";
-    RUN $EARTHLY_FUNCTIONS_HOME/rename-output.sh
+      echo "+CARGO: copying output"; \
+      $EARTHLY_FUNCTIONS_HOME/copy-output.sh "$output"; \
+      echo "+CARGO: removing local crates from target cache"; \
+      cargo metadata --format-version=1  --no-deps | /tmp/jq -r '.packages[].name' | xargs -I{} cargo clean -p {};
```

Comment:
The call to `cargo clean` needs to match the profile used by `cargo $args` e.g. needs to include `--release` or `--profile=foo` as needed. Or I need to figure out a way to say "all profiles".

I'll have a think.

Reply:
I can't find a way to get a list of interesting profiles to clean. Other than "ask the user to supply it" or something ugly involving inspecting `$args` I can't currently see a way to make this approach work.

Reply:
This is further complicated by tools such as [`cargo llvm-cov`](https://github.com/taiki-e/cargo-llvm-cov) which overrides target dir (to `target/llvm-cov-target/`).